scenes/objects/ExplosiveBarrel.gd

```diff
@@ -0,0 +1,38 @@
+extends StaticBody2D
+
+@onready var health := $Health as Health
+@onready var animation := $Explosion/Animation as AnimationPlayer
+@onready var particle_player := $ExplosionParticle/Animation as AnimationPlayer
+@onready var explosion_area := $ExplosionArea as Area2D
+
+@export var damage := 35.0
+@export var impact_force := 20.0
+
+func handle_hit(damager: Node2D, damage):
+	if damager is Bullet:
+		damage = 1
+	health.hit(damage)
+	if health.is_dead():
+		explode()
```

Comment:
I'm saying that because it happen to me when the enemy get a hit from a shotgun for example, he's dying but because the enemy is hitting by multiple bullet, in the is_dead function we are checking if health <= 0 which can be true everytime a bullet hit an enemy just before he is freed

Reply:
It did call the function multiple time but that didn't interfere with the expected barrel behavior. Still took care of it tho